python/ray/util/sgd/v2/trainer.py

```diff
@@ -58,12 +98,16 @@ def run(self,
             callbacks (Optional[List[Callback]]): A list of Callbacks which
                 will be executed during training. If this is not set,
                 currently there are NO default Callbacks.
+
         Returns:
             A list of results from the training function. Each value in the
             list corresponds to the output of the training function from
             each worker.
         """
-        pass
+        config = {} if config is None else config
```

Comment:
Could we check the number of arguments that `train_func` accepts here. You can do that via the `inspect` library. 

What we can do:
- If num_args > 1 then raise ValueError
- If num_args == 1 then continue
- If num_args == 0, then we can wrap it in another function that accepts one arg. So like 
```
if num_args == 0:
    def new_train_fn(config):
        return train_func()

self._executor.run(new_train_fn, config)
```

This way if the user doesn't want to use config we can still support it. What do you think? Alternatively we could also just raise an error if train_func does not accept any args.

Reply:
Yeah this makes sense! I'm wondering if it to just go in the reverse direction so that `BackendExecutor` doesn't need to handle the `config` as well. I think it should functionally be the same...

```python
if num_args == 1: 
    def new_train_fn(): 
        return train_func(config) 

self._executor.run(new_train_fn)
```

One reason we might not want to do this is if `BackendExecutor` will read or populate `config` itself, but I don't currently see a use-case for that.

Reply:
Yep this makes sense!